classes/helper/login_helper.php

```diff
@@ -266,9 +285,11 @@ public function authenticate() {
             if (isguestuser($user)) {
                 // no predefined language for guests - use existing session or default site lang
                 unset($user->lang);
-            } else if (!empty($user->lang)) {
-                // unset previous session language - use user preference instead
-                unset($SESSION->lang);
+            } else {
+                if (!empty($user->lang)) {
```

Comment:
Can we put this back to an elseif to avoid nested if statements please

Reply:
Done, Please review.